data/src/main/java/me/fontys/semester4/data/entity/Product.java

```diff
@@ -1,81 +1,167 @@
 package me.fontys.semester4.data.entity;
 
-import javax.persistence.Column;
-import javax.persistence.Entity;
-import javax.persistence.Id;
-import javax.persistence.Table;
-
+import javax.persistence.*;
+import java.io.Serial;
 import java.io.Serializable;
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Set;
 
 @Entity
 @Table(name = "product")
 public class Product implements Serializable {
 
+    @Serial
     private static final long serialVersionUID = 1L;
 
     @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     @Column(name = "productid", nullable = false)
     private Long productid;
 
-    @Column(name = "taxrate")
-    private Double taxrate;
-
     @Column(name = "name")
     private String name;
 
+    @Column(name = "description")
+    private String description;
+
+    @Column(name = "spicy")
+    private Boolean spicy;
+
+    @Column(name = "vegetarian")
+    private Boolean vegetarian;
+
+    @Column(name = "deliveryfee")
+    private BigDecimal deliveryfee;
+
+    @Column(name = "taxrate")
+    private Double taxrate;
+
     @Column(name = "imagepath")
     private String imagepath;
 
+    @ManyToMany(cascade = {CascadeType.ALL})
+    @JoinTable( name = "category_product",
+            joinColumns = @JoinColumn(name="productid"),
+            inverseJoinColumns = @JoinColumn(name="catid"))
+    private Set<Category> categories;
+
+    @ManyToMany(cascade = {CascadeType.ALL})
+    @JoinTable( name = "ingredient_product",
+            joinColumns = @JoinColumn(name="productid"),
+            inverseJoinColumns = @JoinColumn(name="ingredientid"))
+    private Set<Ingredient> ingredients;
+
+    public Product(Long productid,
+                   String name,
```

Comment:
Splitting the constructor across multiple lines is great for readability when they otherwise exceed the maximum width (in IntelliJ, around ~120 characters), but I wouldn't call it a readability improvement to split every parameter on a new line. Just creates unnecessarily long scrolling sessions 😛

Reply:
Agreed, does IntelliJ also have a quick action for that?